File: RegularReaders/PhoneReader.js
var phoneReg = /(\+?([ .-]?[(]?\d+[)]?)+)/gmi
var text = "+79204441112 \n  89204441112 \n  +7-920-444-11-12 \n +7(920)-444-11-12\n +12308002211  +812308002211"
var phones = text.match(phoneReg)
console.log(phones)

File: RegularReaders/PhoneReader.py
import re

phoneReg = r'(\+7|8).*?(\d{3}).*?(\d{3}).*?(\d{2}).*?(\d{2})'
text = "Мой номер телефона (нет) +79204441112 ну или 89204441112 или 8-920-444-11-12"
phones = [
    m.group()
    for m in re.finditer(phoneReg, text)
]
print(phones)